### ISSUE TITLE
[v.1.10.0] Release Tracker

### ISSUE THREAD

**Issue by malfet** [pytorch/pytorch#65438]
We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/1.10) for the 1.10.0 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 10/7): work on finalizing the release branch
* Phase 2 (after 10/7): perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until 10/7/21):**

Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent minor release (e.g. 1.9 for 1.10 release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)
3. Fixes to new features introduced in the most recent minor release (e.g. 1.9.1 for 1.10 release)
4. Test/CI fixes
5. Documentation improvements
6. Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries
7. Release branch specific changes (e.g. change version identifiers)

Any other change requires special dispensation from the release managers (currently @malfet, @seemethere, @gchanan, @zhouzhuojie ). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

**Phase 2 (after 10/7/21):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/1.9).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/1.10
    git checkout release/1.10
    git cherry-pick abcdef12345

    # Submit a PR based against 'release/1.10' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/1.10
    ```

    </details>
3. Make a request below with the following format:

```
Link to landed master PR (if applicable):
Link to release branch PR:
Criteria Category:
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

See [HUD 1.10](https://hud.pytorch.org/build2/pytorch-release/1.10) or #65434 for release branch CI status

**Comment by malfet** [comment]
Link to landed master PR (if applicable): N_A
Link to release branch PR: https://github.com/pytorch/pytorch/pull/65435
Criteria Category: Releng

**Comment by zou3519** [comment]
Link to landed master PR (if applicable): n/a
Link to release branch PR: https://github.com/pytorch/pytorch/pull/65496
Criteria Category: Release branch specific changes (removing prototype features not ready for release)

**Comment by malfet** [comment]
1.10.0-rc1 build is [here](https://app.circleci.com/pipelines/github/pytorch/pytorch/385188/workflows/7b0c62a4-3c45-49dd-b55b-839753647ce7)

**Comment by prabhat00155** [comment]
Link to landed master PR (if applicable): https://github.com/pytorch/pytorch/pull/65495
Link to release branch PR: https://github.com/pytorch/pytorch/pull/65755
Criteria Category: 2

**Comment by zhouzhuojie** [comment]
Link to landed master PR: N/A
Link to release branch PR: https://github.com/pytorch/pytorch/pull/65787
Criteria Category: 4

----

@zhouzhuojie: merged

**Comment by prabhat00155** [comment]
> Link to landed master PR (if applicable): #65495 Link to release branch PR: #65755 Criteria Category: 2
> 
> @malfet: This does not look like a regression/backward compatibility fix, but a bit more like a feature work: as PR adds new argument to `torch.optim.AveragedModel` class

AveragedModel's update_parameter() doesn't update the parameters correctly in certain cases. This PR fixes that. Look at https://github.com/pytorch/pytorch/pull/65495#issuecomment-929988377 and https://github.com/pytorch/pytorch/pull/65495#pullrequestreview-761703176 for more details.

**Comment by malfet** [comment]
Link to landed master PR: #65721
Link to release branch PR: https://github.com/pytorch/pytorch/pull/65835
Criteria Category: Performance regression from previous PR

**Comment by ejguan** [comment]
Link to landed master PR: #65220
Link to release branch PR: #65924
Criteria Category: Fixes to new features introduced in the most recent minor release (e.g. 1.9.1 for 1.10 release) #64951

**Comment by ejguan** [comment]
Link to landed master PR: #63646
Link to release branch PR: #65926
Criteria Category: Fixes to regressions against the most recent minor release (#63609)

**Comment by NivekT** [comment]
Link to landed master PR: #65424, #65827, #65877
Link to release branch PR: #65932
Criteria Category: Fixes to regressions against the most recent minor release

**Comment by garymm** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/64370 https://github.com/pytorch/pytorch/pull/64371 https://github.com/pytorch/pytorch/pull/64373 https://github.com/pytorch/pytorch/pull/64380 https://github.com/pytorch/pytorch/pull/64382
Link to release branch PR: https://github.com/pytorch/pytorch/pull/65962
Criteria Category: Deprecations of things we want to remove in the next release. Should have no functional change other than logging warnings.

**Comment by malfet** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/65934
Link to release branch PR: https://github.com/pytorch/pytorch/pull/65979
Criteria Category: CI fixes

**Comment by n-v-k** [comment]
Link to landed master PR: #66031
Link to release branch PR: #66117
Criteria Category: 6

**Comment by malfet** [comment]
Link to the landed master PR: #65444
Link to the release branch PR: #66155
Criteria Category: Release regression

**Comment by stas00** [comment]
a likely regression - not triaged - https://github.com/pytorch/pytorch/issues/66086

**Comment by Jianhui-Li** [comment]
Hi, we would like to get Intel Extension to Pytorch tutorial into 1.10 release. Could you please merge it in? 

@vitaly-fedyunin and @gottbrath  give me this link. 

Thanks.

**Comment by malfet** [comment]
> a likely regression - not triaged - #66086

This issue is already have 1.10 milestone

**Comment by malfet** [comment]
> Hi, we would like to get Intel Extension to Pytorch tutorial into 1.10 release. Could you please merge it in?
> 
> @vitaly-fedyunin and @gottbrath give me this link.
> 
> Thanks. @malfet

@Jianhui-Li can you please share link to the PR? Has it been merged into the main branch yet?

**Comment by Jianhui-Li** [comment]
> > Hi, we would like to get Intel Extension to Pytorch tutorial into 1.10 release. Could you please merge it in?
> > @vitaly-fedyunin and @gottbrath give me this link.
> > Thanks. @malfet
> 
> @Jianhui-Li can you please share link to the PR? Has it been merged into the main branch yet?

Sorry to forget the link  https://github.com/pytorch/tutorials/pull/1702

**Comment by malfet** [comment]
Link to the landed master branch PR: N/A
Link to the release branch PR: https://github.com/pytorch/pytorch/pull/66202
Criteria category: Release specific CI fixes

---
@malfet: merged

**Comment by Chillee** [comment]
Link to the landed master branch PR: https://github.com/pytorch/pytorch/pull/59569
Link to the release branch PR: https://github.com/pytorch/pytorch/pull/66213
Criteria category: regression

**Comment by ngimel** [comment]
Link to master branch PR #66191
Link to release branch PR #66214
Criteria category: regression

**Comment by garymm** [comment]
ONNX release notes:
## Backwards Incompatible changes

-   Remove `aten` arg from `onnx.export`. Users can set
    `operator_export_type=OperatorExportTypes.ONNX_ATEN` for
    equivalent behavior
    (<https://github.com/pytorch/pytorch/pull/62759>)

## Deprecations

-   Deprecate several arguments of torch.onnx.export. These will be
    removed in the next release.

    -   enable_onnx_checker
        (<https://github.com/pytorch/pytorch/pull/64369>)

    -   use_external_data_format
        (<https://github.com/pytorch/pytorch/pull/64382>)

    -   example_outputs
        (<https://github.com/pytorch/pytorch/pull/64380>)

    -   strip_doc_string
        (<https://github.com/pytorch/pytorch/pull/64371>)

    -   \_retain_param_name
        (https://github.com/pytorch/pytorch/pull/64370)

## Bug fixes

-   Shape type inference fixes for control flow
    (<https://github.com/pytorch/pytorch/pull/60248>)

-   Fix sum export with attribute keepdims
    (<https://github.com/pytorch/pytorch/pull/60245>)

-   Fix shape inference for large model
    (<https://github.com/pytorch/pytorch/pull/60244>)

-   Fix split export in op set 13
    (<https://github.com/pytorch/pytorch/pull/57605>)

-   Fix control-flow shape inference with contrib op
    (<https://github.com/pytorch/pytorch/pull/62762>)

-   Update instance_norm2d export to handle track_running_stats=True
    (<https://github.com/pytorch/pytorch/pull/58690>)

-   Fix the issue of converting empty list to
    sequence(<https://github.com/pytorch/pytorch/pull/61558>)

-   Fix sum could not be exported for empty tensor.
    (<https://github.com/pytorch/pytorch/pull/59537>)

-   Fix an issue that optimizations might adjust graph inputs
    unexpectedly (<https://github.com/pytorch/pytorch/pull/62763>)

## New Features

-   Add support for ONNX op set 14
    (<https://github.com/pytorch/pytorch/pull/59486>)

-   Support GRU RNNs with packed input in scripting mode
    (<https://github.com/pytorch/pytorch/pull/58691>)

-   Enhanced shape inference provides more accurate shape inference on
    the exported ONNX graph when tracing yields wrong shape inference
    result. Therefore, it increases the exporter coverage of several
    complicated models (<https://github.com/pytorch/pytorch/pull/64585>)

-   Support `torch.linspace`
    (<https://github.com/pytorch/pytorch/pull/58854>)

-   Support `torch.new_ones`
    (<https://github.com/pytorch/pytorch/pull/59255>)

-   Support `torch.nn.LSTMCell`
    (<https://github.com/pytorch/pytorch/pull/62757>)

-   Support `torch.bernoulli` export
    (<https://github.com/pytorch/pytorch/pull/59536>)

-   Support `torch.distributions.normal.Normal` export
    (<https://github.com/pytorch/pytorch/pull/61560>)

-   Support `torch.roll` export
    (<https://github.com/pytorch/pytorch/pull/58697>)

-   Support `torch.dot` and `torch.nn.utils.spectral_norm`
    (<https://github.com/pytorch/pytorch/pull/62765>)

## Improvements

-   Improve error message when attempting to export dict modification
    (`prim::DictConstruct`)
    (<https://github.com/pytorch/pytorch/pull/58696>)

-   Handle onnx::Size in constant folding
    (<https://github.com/pytorch/pytorch/pull/60243>)

-   Update special post process for SequenceInsert after SequenceEmpty
    (<https://github.com/pytorch/pytorch/pull/58693>)

-   Add supplement comments for standardOps low precision cast
    (<https://github.com/pytorch/pytorch/pull/61561>)

-   Fix error message for type_as parameter unknown dtype
    (<https://github.com/pytorch/pytorch/pull/59535>)

## Documentation

-   Major rewrite of documentation
    (<https://github.com/pytorch/pytorch/pull/60249>).

**Comment by cccclai** [comment]
Link to the landed master branch PR: #63421, #64087, #66237
Link to the release branch PR: #66395

Criteria category: It's a feature for 1.10. The change won't affect anything in the desktop/mobile release. User will need to check out the the code in `release/1.10` and build again. It's just to make sure user has a stable branch to check out code and use the feature.

**Comment by kiukchung** [comment]
Hi I'd like to get https://github.com/pytorch/pytorch/pull/66182 cherry-picked, this PR adds hostnames to error summary and improves documentation for `torch.distributed.elastic` (specifically `torch.distributed.run`).

**Comment by malfet** [comment]
pytorch v1.10.0-rc3 https://app.circleci.com/pipelines/github/pytorch/pytorch/394756/workflows/b23151b1-e3f1-4ddd-82f4-5afc61a66364

**Comment by randolf-scholz** [comment]
Did you guys just accidentally release 1.10? It can be installed through conda: 

```      
pytorch                       1.10.0     py3.6_cpu_0  pytorch             
pytorch                       1.10.0 py3.6_cuda10.2_cudnn7.6.5_0  pytorch             
pytorch                       1.10.0 py3.6_cuda11.1_cudnn8.0.5_0  pytorch             
pytorch                       1.10.0 py3.6_cuda11.3_cudnn8.2.0_0  pytorch             
pytorch                       1.10.0     py3.7_cpu_0  pytorch             
pytorch                       1.10.0 py3.7_cuda10.2_cudnn7.6.5_0  pytorch             
pytorch                       1.10.0 py3.7_cuda11.1_cudnn8.0.5_0  pytorch             
pytorch                       1.10.0 py3.7_cuda11.3_cudnn8.2.0_0  pytorch             
pytorch                       1.10.0     py3.8_cpu_0  pytorch             
pytorch                       1.10.0 py3.8_cuda10.2_cudnn7.6.5_0  pytorch             
pytorch                       1.10.0 py3.8_cuda11.1_cudnn8.0.5_0  pytorch             
pytorch                       1.10.0 py3.8_cuda11.3_cudnn8.2.0_0  pytorch             
pytorch                       1.10.0     py3.9_cpu_0  pytorch             
pytorch                       1.10.0 py3.9_cuda10.2_cudnn7.6.5_0  pytorch             
pytorch                       1.10.0 py3.9_cuda11.1_cudnn8.0.5_0  pytorch             
pytorch                       1.10.0 py3.9_cuda11.3_cudnn8.2.0_0  pytorch  
```

However, neither the [releases page](https://github.com/pytorch/pytorch/releases) nor the [1.10.0 milestone](https://github.com/pytorch/pytorch/milestone/24) nor the [1.10 release tracker](https://github.com/pytorch/pytorch/issues/65438) nor [pypi](https://pypi.org/project/torch/#history) indicate the update shipped yet.
